### ISSUE TITLE
[dev] Attempting vanilla "$" in MathQuill.TextField, produces Uncaught ReferenceError

### ISSUE THREAD

**Issue by edborden** [mathquill/mathquill#427]
On the Master branch if I want to type a "$" into the `.mathquill('editable')`, I have to just press "$" twice (once brings me into Math mode, the second brings me out with a text "$").

The same thing on the dev branch in the MathQuill.TextField gets me the following error:

```
Uncaught ReferenceError: dir is not defined
_.createBlocks.ends.(anonymous function).write @ mathquill.js:3022
Controller.open._.typedText @ mathquill.js:1810
typedText @ mathquill.js:1462
(anonymous function) @ mathquill.js:1380
jQuery.event.dispatch @ jquery.js:4430
jQuery.event.add.elemData.handle @ jquery.js:4116
```

**Comment by edborden** [comment]
To clarify, I want to be able to type the character "$" into the TextField.  Somehow.  It seemed like that was possible on master in the most reasonable way, as described above.

**Comment by laughinghan** [comment]
Yeah, `MathQuill.TextField` is deliberately undocumented in the `dev` branch README. You're making a math chat thing, right? I pretty much hacked together `mathquill-textbox` back when we made https://github.com/jneen/lolgebra and since that was before we had tests, it's been barely maintained since. Wanna take charge of this? :D

**Comment by laughinghan** [comment]
Augh, to be honest, once I learned what `$` and `$$` do in Plain TeX (and hence also work in LaTeX though they're discouraged), I realized that was a fuckup and typing `$$` should probably do something else, and `\$` should be the way to type a dollar sign (just like in TeX); or `$` should just type a "$" dollar sign, and there should be some other way to switch between math and text mode.

**Comment by edborden** [comment]
I see...  Also when I render `$x$` via the latex command into the TextField, it changes it on the way in to `$x\$$`.  Bummer, I got really far with the master branch, all this stuff worked correctly, but I wanted the `outOf` handlers in dev, which I guess won't work with TextField anyway

**Comment by laughinghan** [comment]
If you'd rather just hack `master`, the closest equivalent of the `outOf*` handlers would probably be triggering a jQuery custom event in `Cursor::{move,delete,select}Dir`. Something like
```diff
    }
    else {
      // we're at the beginning/end of the containing block, so do nothing
-     if (this.parent === block) return;
+     if (this.parent === block) {
+       if (block === this.root) this.root.jQ.trigger('moveOutOf', dir);
+       return;
+     }

      if (this.parent[dir]) this.insAtDirEnd(-dir, this.parent[dir]);
      else this.insDirOf(dir, this.parent.parent);
```
around https://github.com/mathquill/mathquill/blob/ecb5c3cb5384e92c72514840afe5ed99fea6b082/src/cursor.js#L107 for `moveOutOf`, the others should be even simpler since I think they directly compare with `this.root`.
